pydantic/class_validators.py

```diff
@@ -25,6 +25,13 @@ class Validator:
     always: bool
     check_fields: bool
 
+    def __init__(self, func: AnyCallable, pre: bool, whole: bool, always: bool, check_fields: bool):
```

Comment:
This is necessary due to an issue with Cython failing to set __annotations__, which is used by dataclass.

https://github.com/cython/cython/issues/2552

Reply:
then I guess remove the `@dataclass` decorator and attributes above to avoid duplication.